### ISSUE TITLE
Arch Plasma 6: CMake Error at /usr/lib/cmake/Qt6Qml/Qt6QmlMacros.cmake:2707 (message):

### ISSUE THREAD

**Issue by EmptyVisual** [matinlotfali/KDE-Rounded-Corners#140]
Hi there, I'm unable to compile this on Arch using the KDE Unstable Repo (KDE Plasma 6.0 RC1)

Operating System: Arch Linux 
KDE Plasma Version: 5.92.0
KDE Frameworks Version: 5.248.0
Qt Version: 6.7.0
Kernel Version: 6.7.0-arch3-1 (64-bit)
Graphics Platform: Wayland

Dependencies installed > git cmake extra-cmake-modules base-devel qt5-tools qt6-tools

```
Cloning into 'KDE-Rounded-Corners'...
remote: Enumerating objects: 1643, done.
remote: Counting objects: 100% (197/197), done.
remote: Compressing objects: 100% (62/62), done.
remote: Total 1643 (delta 155), reused 151 (delta 135), pack-reused 1446
Receiving objects: 100% (1643/1643), 4.05 MiB | 10.50 MiB/s, done.
Resolving deltas: 100% (1066/1066), done.
-- The C compiler identification is GNU 13.2.1
-- The CXX compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Installing in the same prefix as Qt, adopting their path scheme.
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE  
-- Found OpenGL: /usr/lib/libOpenGL.so   
-- Found WrapOpenGL: TRUE  
-- Found XKB: /usr/lib/libxkbcommon.so (found suitable version "1.6.0", minimum required is "0.5.0") 
-- Found WrapVulkanHeaders: /usr/include  
CMake Error at /usr/lib/cmake/Qt6Qml/Qt6QmlMacros.cmake:2707 (message):
  qt_generate_foreign_qml_types() is only available in Qt 6.
Call Stack (most recent call first):
  /usr/lib/cmake/Qt6Qml/Qt6QmlConfig.cmake:142 (include)
  /usr/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /usr/lib/cmake/KF6Config/KF6ConfigConfig.cmake:53 (find_dependency)
  /usr/share/ECM/find-modules/FindKF6.cmake:52 (find_package)
  CMakeLists.txt:45 (find_package)


-- Configuring incomplete, errors occurred!
make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.

```

**Comment by matinlotfali** [comment]
I haven't tested the Plasma 6 on Arch Linux.

Looking at your logs, one thing that comes to my mind is the qt5 installation.

Can you remove it and try again?

**Comment by alphabitserial** [comment]
The KDE-Unstable Arch repo currently only provides phonon-qt6-vlc and not phonon-qt6-gstreamer, which pulls in VLC, which pulls in qt5.

**Comment by matinlotfali** [comment]
Can you send me a tutorial to install KDE-Unstable on Arch so I can try it myself?

**Comment by alphabitserial** [comment]
Sure!

#### If you haven't already got Arch installed:
1. Download the install media & flash it to your USB
2. Boot it up and type `archinstall`
3. Select the KDE profile and other pertinent options (you may want to add `firefox` to the list of extra packages to install)

#### Once you're booted into your Arch installation:
Edit /etc/pacman.conf:
1. At the bottom of the file, add these lines above the `[core]` repo entry:
```
[kde-unstable]
Include = /etc/pacman.d/mirrorlist
```
2. In a terminal, run `sudo pacman -Syuuu` and accept any "X package replaces Y package" prompts

**Comment by matinlotfali** [comment]
Thanks for the info. I installed Arch with `kde-unstable` and I found the solution [here](https://www.mail-archive.com/kde-bugs-dist@kde.org/msg697194.html):
> Pass -DQT_MAJOR_VERSION=6 to cmake.

And it worked. Simply run the following commands inside the `build` directory:
```bash
rm -rf ./*
cmake .. --install-prefix /usr -DQT_MAJOR_VERSION=6
```

**Comment by alphabitserial** [comment]
Would you add this information to the readme?

**Comment by matinlotfali** [comment]
I found the fix in #146. It won't need to be added to the readme.
